lib/chef/resource/yum_package.rb

```diff
@@ -62,6 +62,11 @@ class YumPackage < Chef::Resource::Package
       property :allow_downgrade, [ true, false ], default: false
 
       property :yum_binary, String
+
+      def get_package_name(x)
+        return x.to_a if x.is_a?(Array)
```

Comment:
We don't need to call `to_a` on the array. `return x if x.is_a?(Array)` is all we need since the object will already be an array when this evaluates as true.

Reply:
Actually we need to add for ImmutableArray otherwise it's causing issue LIKE:
```
["one", "two", "three"] not to be a kind of Chef::Node::ImmutableArray
```
https://github.com/chef/chef/blob/master/lib/chef/node/immutable_collections.rb#L61